src/components/pill-selector-dropdown/PillSelectorDropdownField.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+import * as React from 'react';
+import getProp from 'lodash/get';
+import type { FieldProps } from 'formik';
+
+import PillSelectorDropdown from './PillSelectorDropdown';
+import defaultDropdownRenderer from './defaultDropdownRenderer';
+import defaultDropdownFilter from './defaultDropdownFilter';
+import parseCSV from '../../utils/parseCSV';
+import type { Option, OptionValue } from './flowTypes';
+
+type Props = FieldProps & {
+    /** CSS class for the component. */
+    className?: string,
+    /** Given selected values and input text, returns a list of filtered options. Defaults to defaultDropdownFilter. */
+    dropdownFilter: (options: Array<Option>, selectedValues: Array<Option>, inputText: string) => Array<Option>,
+    /** Given options, renders the dropdown list. Defaults to defaultDropdownRenderer. */
+    dropdownRenderer: (options: Array<Option>) => React.Node,
+    /** Function to parse user input into an array of items. Defaults to CSV parser. */
+    inputParser: (inputValue: string) => Array<Option>,
+    /** If true, the user can add pills not included in the dropdown options. Defaults to true. */
+    isCustomInputAllowed: boolean,
+    /** If true, the input control is disabled so no more input can be made. Defaults to false. */
+    isDisabled: boolean,
+    /** Pill selector label. */
+    label: React.Node,
+    /** Array of options shown in the dropdown. */
+    options: Array<Option>,
+    /** Called to check if pill text is valid. The text is passed in. */
+    placeholder: string,
+    /** A placeholder to show in the input when there are no pills. */
+    validator?: (option: Option | OptionValue) => boolean,
+};
+
+type State = {
+    inputText: string,
+};
+
+class PillSelectorDropdownField extends React.PureComponent<Props, State> {
+    static defaultProps = {
+        dropdownFilter: defaultDropdownFilter,
+        dropdownRenderer: defaultDropdownRenderer,
+        inputParser: parseCSV,
+        isCustomInputAllowed: true,
+        isDisabled: false,
+        options: [],
+    };
+
+    state = { inputText: '' };
+
+    createFakeEventTarget(name: string, value: Array<Option>) {
+        return { target: { name, value } };
+    }
+
+    handleInput = (text: string) => {
+        this.setState({ inputText: text });
+    };
+
+    handleSelect = (options: Array<Option>) => {
+        const { field } = this.props;
+        const { name, onChange, value = [] } = field;
+        onChange(this.createFakeEventTarget(name, [...value, ...options]));
+    };
+
+    handleRemove = (option: Option, index: number) => {
+        const { field } = this.props;
+        const { name, onChange, value } = field;
```

Comment:
Is `value = []` needed here or not needed in `handleSelect`?

Reply:
handleRemove gets called when you click on the X button on a pill which is generated using value, so in that case the value should always be there